client/src/App.js

```diff
@@ -44,16 +44,22 @@ axios.interceptors.response.use(
   (response) => response,
   async (error) => {
     const originalRequest = error.config;
+    if (originalRequest.url === "/api/v1/users/logout") {
+      // to avoid loop
+      return Promise.reject(error);
+    }
+
     let { refreshToken } = AuthService.getCurrentUser() || {};
```

Comment:
Not related with your changes but still

```suggestion
    const { refreshToken } = AuthService.getCurrentUser() || {};
```